### PR TITLE
Add the ability to toggle camera mode backwards

### PR DESCRIPTION
**Pull request by bigfoot547**
Before, you could only go foreword (first person, third person, third person front) but now you can go backwards! Press `sneak+F7` to go backwards. Please tell me if you should use the `e` key instead.

### CODE REVIEW

**Comment by stujones11** [comment]
Interesting idea but I think, if accepted, it should work with all function keys that sequence through multiple options.

**Comment by bigfoot547** [comment]
That could be arranged...

**Comment by bigfoot547** [comment]
Why don't you like it @kilbith ?

**Comment by paramat** [comment]
:-1: Sorry but with only 3 options going backwards is not needed, locating and pressing another key together with F7 to go back 1 would be more bother than cycling forwards 2 with 2 F7 presses.

**Comment by bigfoot547** [comment]
:-(

**Comment by Calinou** [comment]
I do see an use for disabling the front-view third person mode entirely using a configuration setting, mostly for PvP players who may want to switch between modes quickly, and couldn't care less about the front-view third person mode.

If you think this is worthwhile, I can create an issue about it.

**Comment by bigfoot547** [comment]
I personally don't ever use 3<sup>rd</sup> person mode from the back, what are the usecases for it anyway?

**Comment by paramat** [comment]
bigfoot you don't have to close this, other devs may like it.

**Comment by sofar** [comment]
This is a little bit too complex for my taste.

But, if you added over-the-shoulder 3rd person view, then maybe something like this would make more sense.

**Comment by DS-Minetest** [comment]
> over-the-shoulder 3rd person view

Or a view where if you move mouse not the player but only the camera turns.

**Comment by wilkgr76** [comment]
@DS-Minetest Actually, in multiplayer, it would be better for the player to turn, esp. in PvP scenarios.

File: src/camera.h [hunk 1]
```diff
@@ -161,6 +161,15 @@ class Camera
 			m_camera_mode = CAMERA_MODE_FIRST;
 	}
 	
+	void toggleCameraModeBackwards() {
+		if (m_camera_mode == CAMERA_MODE_FIRST)
```
Comment (nerzhul) [inline src/camera.h]: switch case

File: src/game.cpp [hunk 2]
```diff
@@ -3411,8 +3412,19 @@ void Game::updateCamera(u32 busy_time, f32 dtime)
...
+		if (playercao == NULL)
+			return;
+		
+		camera->toggleCameraModeBackwards();
+		
+		playercao->setVisible(camera->getCameraMode() > CAMERA_MODE_FIRST);
+		playercao->setChildrenVisible(camera->getCameraMode() > CAMERA_MODE_FIRST);
 	}
-
+
```
Comment (nerzhul) [inline src/game.cpp]: no

**Comment by DS-Minetest** [comment]
@wilkgr76 Or both.

**Comment by paramat** [comment]
Disapproval from rubenwardy http://irc.minetest.net/minetest-dev/2017-08-05#i_5034348

**Pull request closed without merging**